### PR TITLE
Graphql

### PR DESCRIPTION
**Pull request by karimkawambwa**
This branch will serve as a staging branch for introducing graphql to this project.

**NOTE:** Any other graphql work **should branch off from this branch** and **merge to this branch** and then **this branch will be merged to master.**

### CODE REVIEW

File: src/pages/Profile.js [hunk 1]
```diff
@@ -1,87 +1,258 @@
...
+  // Provide the visuals with unique ids for fetching
+  // The unique ids will be used to set alias in graphql
+  let index = 0;
+  sectionedCharts.forEach(x =>
+    x.charts.forEach(y =>
+      y.visuals.forEach(z => {
+        // eslint-disable-next-line no-param-reassign
+        z.id = `viz${index}`;
+        index += 1;
+      })
```
Comment (KhadijaMahanga) [inline src/pages/Profile.js]: why don't we have unique ids defined in the json file?

File: src/pages/Profile.js [hunk 2]
```diff
@@ -1,87 +1,258 @@
...
+      .map(
+        visual => `${visual.id}: ${visual.table} (
+      condition: { geoCode: $geoCode, geoLevel: $geoLevel }
+    ) {
+      nodes {
+        ${
+          visual.label && visual.label[0] === '$'
+            ? `label: ${visual.label.slice(1)}`
+            : ''
+        }
```
Comment (KhadijaMahanga) [inline src/pages/Profile.js]: I don't understand the dollar sign on label.

From the if statement, If we have a defined label and first letter is not '$', we do not set a label at all. Is this correct?

File: src/pages/Profile.js (hunk 1 above)
Comment (karimkawambwa) [reply src/pages/Profile.js]: I decided to do this to reduce the amount of variables we set on the json file. I also think this way we can make the ids truly unique so we don't confuse when making the alias for the chart data. What are your thoughts @KhadijaMahanga ? You think we should go back to just specify them?

 cc @DavidLemayian

File: src/pages/Profile.js (hunk 2 above)
Comment (karimkawambwa) [reply src/pages/Profile.js]: So if we specify it with a dollar sign. I take it as a variable we go to the db to fetch. Example is the "$longName". I left it there as an example. We can fetch the geo name by just doing that. If we had more relationships, this label could be made to fetch the parent or profile geo name for example, without us having to prefetch the parent geo and/or profile geo.

So if there is no dollar sign i assume you want to query `longName` from the prefetched profiles. If the label does not exist in that profile then I assume the lable is just a string you want to use e.g. `label: "People"`

@KhadijaMahanga

Comment (karimkawambwa) [reply src/pages/Profile.js]: I know its a bit of a circle but since we want default values, we have it working like this.

**Comment by nerdlet** [comment]
@karimkawambwa LGTM, apart from the bug on the Kenya profile.

**Pull request merged**